flexinput/src/main/java/com/lytefast/flexinput/adapters/PhotoCursorAdapter.kt

```diff
@@ -89,10 +89,13 @@ class PhotoCursorAdapter(private val contentResolver: ContentResolver,
 
   private operator fun get(position: Int): Photo {
     cursor!!.moveToPosition(position)
-    val photoId = Integer.toString(cursor!!.getInt(colId))
-    val fileUri = Uri.withAppendedPath(MediaStore.Images.Media.EXTERNAL_CONTENT_URI, photoId)
+    val photoId = cursor!!.getLong(colId)
+    val fileUri = Uri.withAppendedPath(MediaStore.Images.Media.EXTERNAL_CONTENT_URI, photoId.toString())
     return Photo(
-        cursor!!.getLong(colId), fileUri, cursor!!.getString(colName), cursor!!.getString(colData))
+        id = photoId,
+        uri = fileUri,
+        displayName = cursor!!.getString(colName)?: "img-$photoId",
```

Comment:
Same comment about `!!` here and below.  Maybe this whole block should just `apply` or `let` on the `cursor` since that is the source of nullability.

Reply:
You'll have to change some caller code slightly but this method can be made safe w/ `let`:

```kotlin
  private operator fun get(position: Int): Photo? = cursor?.let {
    it.moveToPosition(position)
    val photoId = it.getLong(colId)
    val fileUri = Uri.withAppendedPath(MediaStore.Images.Media.EXTERNAL_CONTENT_URI, photoId.toString())
    return Photo(
            id = photoId,
            uri = fileUri,
            displayName = it.getString(colName)?: "img-$photoId",
            photoDataLocation = it.getString(colData))
  }
```

Then above when you bind the view:

```kotlin
this[position]?.apply { holder.bind(this) }
```

and finally handling the potentially null id:

```kotlin
  override fun getItemId(position: Int): Long = this[position]?.id ?: -1
```

Reply:
I'm trying to prevent this method from returning null. However maybe I'm being too strict for a UI rendering component... I'll make the null checks.

Reply:
the holder check is a bit more complicated and requires more than just not binding since it can display old data from when it was previously bound. I'll have to create an EMPTY style Photo